SU2_CFD/src/solvers/CTurbSASolver.cpp

```diff
@@ -1391,137 +1385,141 @@ void CTurbSASolver::SetDES_LengthScale(CSolver **solver, CGeometry *geometry, CC
+    const su2double fv1  = Ji_3/(Ji_3+cv1_3);
+    const su2double fv2 = 1.0 - Ji/(1.0+Ji*fv1);
+    const su2double ft2 = ct3*exp(-ct4*Ji_2);
+    const su2double cw1 = cb1/k2+(1.0+cb2)/sigma;
 
-    psi_2 = (1.0 - (cb1/(cw1*k2*fw_star))*(ft2 + (1.0 - ft2)*fv2))/(fv1 * max(1.0e-10,1.0-ft2));
+    su2double psi_2 = (1.0 - (cb1/(cw1*k2*fw_star))*(ft2 + (1.0 - ft2)*fv2))/(fv1 * max(1.0e-10,1.0-ft2));
     psi_2 = min(100.0,psi_2);
 
+    su2double lengthScale = 0.0;
+
     switch(kindHybridRANSLES){
-      case SA_DES:
+      case SA_DES: {
         /*--- Original Detached Eddy Simulation (DES97)
         Spalart
         1997
         ---*/
 
-        maxDelta = geometry->nodes->GetMaxLength(iPoint);
-        distDES = constDES * maxDelta;
+        const su2double maxDelta = geometry->nodes->GetMaxLength(iPoint);
+        const su2double distDES = constDES * maxDelta;
         lengthScale = min(distDES,wallDistance);
 
         break;
-
-      case SA_DDES:
+      }
+      case SA_DDES: {
         /*--- A New Version of Detached-eddy Simulation, Resistant to Ambiguous Grid Densities.
          Spalart et al.
          Theoretical and Computational Fluid Dynamics - 2006
          ---*/
 
-        maxDelta = geometry->nodes->GetMaxLength(iPoint);
+        const su2double maxDelta = geometry->nodes->GetMaxLength(iPoint);
 
-        r_d = (kinematicViscosityTurb+kinematicViscosity)/(uijuij*k2*pow(wallDistance, 2.0));
-        f_d = 1.0-tanh(pow(8.0*r_d,3.0));
+        const su2double r_d = (kinematicViscosityTurb+kinematicViscosity)/(uijuij*k2*pow(wallDistance, 2.0));
+        const su2double f_d = 1.0-tanh(pow(8.0*r_d,3.0));
 
-        distDES = constDES * maxDelta;
+        const su2double distDES = constDES * maxDelta;
         lengthScale = wallDistance-f_d*max(0.0,(wallDistance-distDES));
 
         break;
-      case SA_ZDES:
+      }
+      case SA_ZDES: {
         /*--- Recent improvements in the Zonal Detached Eddy Simulation (ZDES) formulation.
          Deck
          Theoretical and Computational Fluid Dynamics - 2012
          ---*/
 
-        for (iNeigh = 0; iNeigh < nNeigh; iNeigh++){
-            jPoint = geometry->nodes->GetPoint(iPoint, iNeigh);
-            coord_j = geometry->nodes->GetCoord(jPoint);
-            for ( iDim = 0; iDim < nDim; iDim++){
-              deltaAux = abs(coord_j[iDim] - coord_i[iDim]);
-              delta[iDim] = max(delta[iDim], deltaAux);
-            }
-            deltaDDES = geometry->nodes->GetMaxLength(iPoint);
+        const su2double deltaDDES = geometry->nodes->GetMaxLength(iPoint);
+
+        su2double delta[MAXNDIM] = {}, ratioOmega[MAXNDIM] = {};
+
+        for (const auto jPoint : geometry->nodes->GetPoints(iPoint)) {
+          const auto coord_j = geometry->nodes->GetCoord(jPoint);
+          for (auto iDim = 0u; iDim < nDim; iDim++){
+            const su2double deltaAux = abs(coord_j[iDim] - coord_i[iDim]);
+            delta[iDim] = max(delta[iDim], deltaAux);
```

Comment:
I currently use gnu-5.3.0 on our cluster which gives me an `array subscript is above array bounds`-warning for the `max(delta[iDim]...`. [This](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=59124) makes me believe that it is problem maybe including this version number.

using `unsigned short` instead of `auto` fixes this unjustified warning. I know it is not an actual problem but I would change that over at #1382 if ok